src/Apps/Order/Routes/Offer/index.tsx

```diff
@@ -110,6 +111,7 @@ export class OfferRoute extends Component<OfferProps, OfferState> {
 
   onMutationError(errors, errorModalTitle?, errorModalMessage?) {
     logger.error(errors)
+    sendErrorToService(new Error(errors.code), errors)
```

Comment:
Why not add a config argument to `logger.error` and send there? By circumventing the logger to deal with localized issue it creates drift for other devs leading to less error coverage. Could do something like 

```
logger.error(errors, { sendErrorToService: true /* the default /* })
```
which then wraps the above call-set. (Not suggesting the above exactly, but you get what I mean :) With the way its been refactored there are now three `errors` objects being passed about which increases complexity and mental overhead, essentially crippling the function of the logger. If a user is already importing the logger it should be assumed to be the canonical  location for extending functionality, regardless of local conditions, and extended via configuration.

Reply:
Hmm, I don't totally follow you — could give a more detailed example of how a config would help here? The problem from my perspective is that we want to send everything through the logger but the logger can’t always determine what it should use to call `sendErrorToService`, and it doesn’t seem like a config would help that at first glance.

Reply:
I'm still missing something, here. How does this help with the problem that we can't figure out with certainty which input contains error metadata?

Reply:
I think @damassi 's approach makes sense too-- but @williardx what do you mean by your last point?

Reply:
Here's a better description of the input disambiguation problem I've been mentioning: 

A mutation fails, and we get back a response that looks like the following: `errors = { code: 'some_code', type: 'some_type', data: { ... } }`. In order for the logger to recognize this as an error, we need to wrap this in an `Error` object. However, `Error` doesn't automatically support adding the extra metadata that we have for this error -- its arguments are a message, filename and line number. 

So, we can create a new error, say `Error(errors.code)`, and the logger will find the error. We pass in the rest of metadata as a separate "error info" object (`errorInfo`), giving us: `logger.error(new Error(errors.code), errorInfo)`.

However, the logger doesn't have any restrictions as to what input it can take. We can find an error by checking for objects that are an instance of `Error`, but we can't do the same for our error info object since it's just a plain object. Someone could end up passing in multiple objects to the logger: `logger.error(new Error(errors.code), errorInfo, "some message", someOtherObj)`. There's no way to know that `errorInfo` contains the metadata we want and that `someOtherObj` doesn't because we don't know ahead of time what `errorInfo` will contain. I don't know why we might have multiple objects there, but I don't want to discount the possibility or restrict the logger. (Previously we could tell because we were looking for objects that implemented a particular `ErrorInfo` interface, but it turns out we don't need that and that we should just let error info objects be generic.)

From what I see, adding a config doesn't actually solve ☝️because knowing whether or not we need to send an error isn't the problem -- it's figuring out if there's metadata to associate with the error.

Reply:
It sounds like we need to define a `Error` subclass for these mutation errors that override the `message` property to return the right information. The `logger.error` implementation shouldn’t need to care about anything else than the `message` property.

Here’s an example I recently wrote https://github.com/1stdibs/relay-mock-network-layer/blob/master/RelayMockNetworkLayerError.js

Reply:
Do you think this `Error` subclass would just be for mutation errors, or should it be used more broadly? If `sendErrorToService` receives some errors that contain metadata and some that have it as a separate object it gets confusing, so I would think we'd want some new general error class that can be used anytime we want to attach metadata to an error. 

This is scenario (3) in the PR description. The `ReactionError` class could be simple -- I don't _think_ we need to override the message:

```
class ReactionError extends Error { 
  metadata: object
  constructor(message, metadata = {}) {
    super(message)
    this.metadata = metadata
  }
}
```

We could then nix the `errorInfo` object in `sendErrorToService` and just pull from `error.metadata` if that property exists:

```
export const sendErrorToService = (error: Error | ReactionError) => {
  Sentry.withScope(scope => {
    if (error instanceof ReactionError) {    
      Object.keys(error.metadata).forEach(key => {
        scope.setExtra(key, error.metadata[key])
      })
    }
    Sentry.captureException(error)
  })
}
```

There's a little weirdness in `componentDidCatch` in `ErrorBoundary` since we get an error and immediately convert it into `ReactionError`, but thinking again that seems all right.